### PR TITLE
LDSW & BUCK: split GPIO cmds

### PR DESCRIPTION
**Pull request by masz-nordic**
test:PR-202

### CODE REVIEW

File: drivers/npmx/shell/ldsw.c [hunk 1]
```diff
@@ -115,32 +129,43 @@ static int cmd_ldsw_gpio_set(const struct shell *shell, size_t argc, char **argv
...
 	if (!check_error_code(shell, err_code)) {
 		print_set_error(shell, "GPIO config");
 		return 0;
 	}
 
-	shell_print(shell, "Success: %d %d.", gpio_index, gpio_config.inverted);
+	int16_t *p_value = (param == LDSW_GPIO_PARAM_INDEX) ? (int16_t *)&gpio_config.gpio :
+							      (int16_t *)&gpio_config.inverted;
+
+	shell_print(shell, "Success: %d.", *p_value);
```
Comment (wilu-nordic) [inline drivers/npmx/shell/ldsw.c:163]: Should be handled with a `switch-case`.

[example](https://github.com/NordicSemiconductor/npmx-zephyr/blob/develop/drivers/npmx/npmx_shell.c#L2676)

File: drivers/npmx/shell/ldsw.c [hunk 2]
```diff
@@ -97,15 +103,23 @@ static npmx_ldsw_gpio_t ldsw_gpio_index_convert(int32_t gpio_idx)
...
-					  [0] = { SHELL_ARG_TYPE_UINT32_INDEX, "LDSW" },
-					  [1] = { SHELL_ARG_TYPE_INT32_VALUE, "GPIO number" },
-					  [2] = { SHELL_ARG_TYPE_BOOL_VALUE,
-						  "GPIO inversion status" },
-				  } };
+	args_info_t args_info = { .expected_args = 2,
+							.arg = {
+							[0] = { SHELL_ARG_TYPE_UINT32_INDEX, "LDSW" },
+							},
+	};
```
Comment (magp-nordic) [inline drivers/npmx/shell/ldsw.c:113]: It looks like something's wrong with the alignment here

File: drivers/npmx/shell/ldsw.c (hunk 1 above)
Comment (magp-nordic) [inline drivers/npmx/shell/ldsw.c:163]: `print_success` should be used.
Also in Buck commit

File: drivers/npmx/shell/ldsw.c [hunk 3]
```diff
@@ -162,13 +187,38 @@ static int cmd_ldsw_gpio_get(const struct shell *shell, size_t argc, char **argv
...
+
+static int cmd_ldsw_gpio_index_get(const struct shell *shell, size_t argc, char **argv)
+{
+	return ldsw_gpio_get(shell, argc, argv, LDSW_GPIO_PARAM_INDEX);
+}
+
+static int cmd_ldsw_gpio_polarity_get(const struct shell *shell, size_t argc, char **argv)
+{
+	return ldsw_gpio_get(shell, argc, argv, LDSW_GPIO_PARAM_POLARITY);
+}
```
Comment (magp-nordic) [inline drivers/npmx/shell/ldsw.c:221]: Wrong ordering, should be:
- `cmd_ldsw_gpio_index_set`
- `cmd_ldsw_gpio_index_get`
- `cmd_ldsw_gpio_polarity_set`
- `cmd_ldsw_gpio_polarity_get`
To match the ordering of commands' creation.
Also in Buck commit

File: drivers/npmx/shell/ldsw.c [hunk 4]
```diff
@@ -162,13 +187,38 @@ static int cmd_ldsw_gpio_get(const struct shell *shell, size_t argc, char **argv
...
-	int8_t gpio_index =
-		(gpio_config.gpio == NPMX_LDSW_GPIO_NC) ? -1 : ((int8_t)gpio_config.gpio - 1);
+	if (param == LDSW_GPIO_PARAM_INDEX) {
+		int8_t gpio_index = (gpio_config.gpio == NPMX_LDSW_GPIO_NC) ?
+					    -1 :
+					    ((int8_t)gpio_config.gpio - 1);
+
+		shell_print(shell, "Value: %d", gpio_index);
+	} else if (param == LDSW_GPIO_PARAM_POLARITY) {
+		shell_print(shell, "Value: %d", gpio_config.inverted);
```
Comment (magp-nordic) [inline drivers/npmx/shell/ldsw.c:197]: `print_value` should be used.

Comment (magp-nordic) [reply drivers/npmx/shell/ldsw.c:197]: Also in Buck commit

File: drivers/npmx/shell/ldsw.c (hunk 1 above)
Comment (masz-nordic) [reply drivers/npmx/shell/ldsw.c:163]: There's no logic here, why is `switch` needed?

File: drivers/npmx/shell/buck.c [hunk 5]
```diff
@@ -90,44 +113,59 @@ static int buck_gpio_set(const struct shell *shell, size_t argc, char **argv,
...
 
-	npmx_error_t err_code = gpio_config_set(buck_instance, &gpio_config);
+	err_code = gpio_config_set(buck_instance, &gpio_config);
 	if (!check_error_code(shell, err_code)) {
 		print_set_error(shell, "GPIO config");
 		return 0;
 	}
 
-	print_success(shell, gpio_index, UNIT_TYPE_NONE);
+	int16_t *p_value = (config_type == BUCK_GPIO_PARAM_INDEX) ?
```
Comment (magp-nordic) [inline drivers/npmx/shell/buck.c:157]: Why pointer? Why not just a variable? Won't casting pointer to bool (`inverted` field) to a pointer to `uint16_t` cause any problems?
Same in LDSW commit

Comment (masz-nordic) [reply drivers/npmx/shell/buck.c]: Fixed

File: drivers/npmx/shell/ldsw.c (hunk 1 above)
Comment (wilu-nordic) [reply drivers/npmx/shell/ldsw.c]: RIght.

**Comment by wilu-nordic** [comment]
Is there a test PR for this PR? @nordic-segl

File: drivers/npmx/shell/buck.c [hunk 6]
```diff
@@ -70,17 +76,34 @@ static int cmd_buck_active_discharge_get(const struct shell *shell, size_t argc,
...
+	args_info_t args_info = { .expected_args = 2,
 				  .arg = {
 					  [0] = { SHELL_ARG_TYPE_UINT32_INDEX, "buck" },
-					  [1] = { SHELL_ARG_TYPE_INT32_VALUE, "GPIO number" },
-					  [2] = { SHELL_ARG_TYPE_BOOL_VALUE,
-						  "GPIO inversion status" },
-				  } };
+					  [1] = { arg_type, config_name },
+					  },
+	};
```
Comment (wilu-nordic) [inline drivers/npmx/shell/buck.c:105]:
```suggestion
					  } };
```

Please remove `,`

File: drivers/npmx/shell/buck.c (hunk 5 above)
Comment (magp-nordic) [reply drivers/npmx/shell/buck.c]: It is still cast to pointer to `uint16_t` in  LDSW.

Comment (masz-nordic) [reply drivers/npmx/shell/buck.c]: Fixed there as well.

File: drivers/npmx/shell/buck.c (hunk 6 above)
Comment (masz-nordic) [reply drivers/npmx/shell/buck.c]: Done

**Review by magp-nordic — DISMISSED** [review]

**Review by magp-nordic — DISMISSED** [review]

**Comment by nordic-segl** [comment]
please note that integration CI is not testing npmx ldsw gpio * (tests are not ready)

**Review by nordic-segl — APPROVED** [review]

**Comment by nordic-segl** [comment]
All tests are passing - OK to merge.

**Review by wilu-nordic — APPROVED** [review]

**Pull request merged**